modules/apps/frontend-js/frontend-js-test/src/testFunctional/macros/Walkthrough.macro

```diff
@@ -78,26 +91,39 @@ definition {
 
 	macro setupWalkthroughPage {
 		task ("Given there is a site-level JSON defined for current site") {
-			JSONGroup.addGroup(groupName = "Site Name");
-
-			ApplicationsMenu.gotoSite(site = "Site Name");
+			if (!(isSet(widgetList))) {
+				var widgetList = "JS Walkthrough Sample";
+			}
 
 			Walkthrough.addWalkthrough(fileName = "${filename}");
 
 			JSONLayout.addPublicLayout(
-				groupName = "Site Name",
-				layoutName = "Walkthrough Page");
+				groupName = "${siteName}",
+				layoutName = "Walkthrough Page 1");
 
 			JSONLayout.addWidgetToPublicLayout(
-				groupName = "Site Name",
-				layoutName = "Walkthrough Page",
+				groupName = "${siteName}",
+				layoutName = "Walkthrough Page 1",
 				widgetName = "JS Walkthrough Sample");
+
+			if ("${filename}" == "walkthrough_configuration_multiple_page.json") {
+				JSONLayout.addPublicLayout(
+					groupName = "${siteName}",
+					layoutName = "Walkthrough Page 2");
+
+				for (var portletName : list "${widgetList}") {
+					JSONLayout.addWidgetToPublicLayout(
+						groupName = "${siteName}",
+						layoutName = "Walkthrough Page 2",
+						widgetName = "${portletName}");
```

Comment:
I don't like this for loop in setup walkthrough macro. We should just deploy walkthrough sample portlet. Adding an extra portlet is for 1 edge case and isn't worth including in this macro. General rule: If there's 3 or more instances of this then it can be considered to include for a new macro or refactoring a macro.

Reply:
Doesn't look like this change was made